Python/Product/TestAdapter.Executor/UnitTest/TestExecutorUnitTest.cs

```diff
@@ -200,27 +196,10 @@ sealed class TestRunner : IDisposable {
 
                 _env = new Dictionary<string, string>();
 
-                _debugMode = PythonDebugMode.None;
-                if (runContext.IsBeingDebugged && _app != null) {
-                    _debugMode = settings.EnableNativeCodeDebugging ? PythonDebugMode.PythonAndNative : PythonDebugMode.PythonOnly;
-                }
-
                 _searchPaths = GetSearchPaths(tests, settings);
 
-                if (_debugMode == PythonDebugMode.PythonOnly) {
-                    if (_settings.UseLegacyDebugger) {
-                        var secretBuffer = new byte[24];
-                        RandomNumberGenerator.Create().GetNonZeroBytes(secretBuffer);
-                        _debugSecret = Convert.ToBase64String(secretBuffer)
-                            .Replace('+', '-')
-                            .Replace('/', '_')
-                            .TrimEnd('=');
-                    } else {
-                        _debugSecret = "";
-                    }
+                ExecutorService.GetDebugSettings(_app, _context, _settings, out _debugMode, out _debugSecret, out _debugPort);
 
-                    SocketUtils.GetRandomPortListener(IPAddress.Loopback, out _debugPort).Stop();
```

Comment:
I think GetRandomPortListener().stop() is still needed right?

Reply:
it's in ExecutorService.GetDebugSettings, it doesn't show in the diff.